stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -856,3 +856,64 @@ for func in (:log, :sqrt)
         end
     end
 end
+
+"""
+    symmetricpart(A, uplo=:U)
+
+Return the symmetric part of the square matrix `A`, defined as `(A + transpose(A)) / 2`,
+as a [`Symmetric`](@ref) matrix.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+symmetricpart(A::AbstractMatrix{T}, uplo=:U) where {T} =
+    symmetricpart!(copyto!(similar(A, typeof(one(T) / 2)), A), uplo)
+
+"""
+    symmetricpart!(A, uplo=:U)
+
+Overwrite the square matrix `A` with its symmetric part, `(A + transpose(A)) / 2`,
+and return `Symmetric(A, uplo)`.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+symmetricpart!(A::AbstractMatrix, uplo::Symbol=:U) =
+    Symmetric(_hermorsympart!(identity, identity, A, char_uplo(uplo)), uplo)
```

Comment:
To be more general, this should be `_hermorsympart!(symmetricpart, transpose, ...)`.  Similarly,  for `hermitianpart` you should use `_hermorsympart!(hermitianpart, adjoint, ...)`.

Then have base cases
```
symmetricpart(x::Number) = x
hermitianpart(x::Number) = real(x)
```